### DISCUSSION TITLE
Can you still host the bot on rebornhosting?

### DISCUSSION THREAD

**Discussion by Deshunsgodly** [jagrosh/MusicBot#903] [Questions] [answered]
If so how I can't figure out how to get the bot to start to get hosted from the website

**Comment by YJSoft** [comment]
You will need paid plan since on free plan, bot will crash due to memory limit)

I assume that you already buy paid java plan.(it's not that hard. I tried)

1. Go to panel (https://botnode1.rebornhosting.net/)
2. Login with your id
3. Upload JMusicBot's jar file with File Manager(Drag & drop or click upload)
4. Rename jar file to bot.jar
5. Click New file, and fill information using [Guide](https://jmusicbot.com/setup#3%EF%B8%8F%E2%83%A3-edit-the-config-file)
6. Save file as config.txt
7. Click start. If all config value are correct, bot will be activated soon.
8. Click invite link shown to console, or follow [Guide](https://jmusicbot.com/adding-your-bot) to add bot to server

p.s.) I'm not promoting any services, but providing information to others.

**Reply by MichailiK** [reply]
> since bot itself consumes about 200MB of ram

I believe Java should be able to deal with the increased memory pressure. The reason the MusicBot/Java consumes 200+ MB is because *it can*. Java prefers to allocate more memory than cleaning up memory (aka. garbage collection), since garbage collection can result in stutters.

**Reply by YJSoft** [reply]
Free servers are out of stock anyway so only option will be paid one.(unless free servers are available)

p.s.) Looks like they just kill process when memory exceed limit(tested with free discord.py hosting). If same limits are applied to free java bot hosting, java would consume more memory than allowed, resulting bot killed.

**Reply by MichailiK** [reply]
You can somewhat tell the JVM how much memory to consume with the `-Xmx` and `-Xms` parameter.

**Reply by YJSoft** [reply]
@MichailiK parameters cannot be changed.(only filename)

**Reply by YJSoft** [reply]
Turns out, it's still possible to run JMusicBot on free plan(although memory is limited to 128M, jar files can use up to 140M)
[Screenshot of dashboard](https://puu.sh/IdzYi/8a063d4c52.png)

Edit) Running bot is possible, but bot will crash due to memory limit. It's weird because they offer one click JMusicBot install...

**Reply by Deshunsgodly** [reply]
@YJSoft After I "Saveed the file as config.txt" and start it img etting thish

![adaac](https://user-images.githubusercontent.com/91224768/135771085-4478a54d-1b21-4124-af3e-4af0ecba74cd.JPG)

**Reply by MichailiK** [reply]
It looks like you're just running `java --version`.

**Reply by YJSoft** [reply]
Check `Startup` -> `JAR FILE` and make sure that you have right filename. default value are `bot.jar` and If you never modified uploaded file name, that should be like `JMusicBot-0.3.6.jar`.

Or you can modify filename to `bot.jar` and bot will start working.

**Comment by YJSoft** [comment, marked as answer]
FYI, rebornhosting and it's botnode is now closed(without any notice)
![image](https://user-images.githubusercontent.com/5893514/148157742-43367646-066b-474f-9b58-a46879b6900b.png)
